src/Raven.Server/Documents/Sharding/ShardedDatabaseContext.cs

```diff
@@ -114,6 +115,19 @@ public void UpdateDatabaseRecord(RawDatabaseRecord record, long index)
             RachisLogIndexNotifications.NotifyListenersAbout(index, e: null);
         }
 
+        public void RefreshExecutors()
+        {
+            using (var se = ShardExecutor)
+            {
+                ShardExecutor = new ShardExecutor(ServerStore, _record, _record.DatabaseName);
+            }
+
+            using (var ne = AllOrchestratorNodesExecutor)
```

Comment:
Is this safe to do? If I have a reference to the old instance, you'll dsipose that midway through, no?

Reply:
We pretty much have to do that with a finalizer, yes.
Note that the executor is using unmanaged memory, so we may get access violation from this behavior.

The only scenario where it is valid to dispose it when we are actually shutting down, and presumably already ensured that there are no active requests using it.

Reply:
Okay, added finalizer.

But, I think the dispose of the executor is problematic in general... I don't see any mechanism to drain all ongoing requests before disposing the it.
Do I miss something? or should I open an issue to handle that?